### PR TITLE
Clean parse_lombardi; Rename model config dir

### LINKED ISSUE #15: Changing Italy stationary model investsteps not filtering through to all CSV files
### What happened?

Some Italy stationary CSV files are static, i.e. not generated by the preprocessing script. If one updates constants in the preprocessing script (e.g., investstep resolution to 5 years), these static CSVs don't update appropriately

### Which operating systems have you used?

MacOS

### Version

v0.1.0

### Relevant log output

_No response_

### LINKED ISSUE #16: Investsteps and vintagesteps are the wrong way around
### What happened?

In the available vintage CSVs, `Investsteps` and `vintagesteps` are the wrong way around (e.g., you can build a `2030` vintage in `2020` investstep). This is caused by the order of columns when setting year pairs in `parse_available_vintages` in `parse_lombardi.py`.

### Which operating systems have you used?

macOS

### Version

v0.1.0

### Relevant log output

_No response_

### LINKED ISSUE #17: Demand available vintages messed up
### What happened?

It should not be possible / desirable to have "vintages" for demand technologies. This leads to `flow_cap` for demand increasing in each investment step, even though peak demand remains constant. I suspect the best approach right now is to set an `initial_flow_cap` and then not allow any new flow cap.

### Which operating systems have you used?

macOS

### Version

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by brynpickering**
Fixes #17 
Fixes #16 
Fixes #15 

A number of changes here:

1. I've made the preprocessing an automated part of loading the italian model. It makes it possible to set custom investment step resolutions / periods and have that information filter through.
2. I've removed some CSV files in favour of defining the data in YAML (where it is the same value in every investment period).
3. I've brought in generating some simple CSVs into the pre-processing step so that changes in investment resolution is passed to them.
4. I've changed from `models` to `model_configs` to not get name clashes with the `models.py` file.
5. I've cleaned up the import and export of electricity from outside italy so that it is actually used by the model. This entails adding custom math to link two technologies.

This does leave `parse_ppm.py` by the wayside, but that's fine for now I think. It's still possible to run it from the command line (as it is with `parse_lomabrdi.py`).

### CODE REVIEW

**Comment by irm-codebase** [comment]
I agree on leaving `ppm` on the side. I was trying to get it to work so that we could change the start year (i.e., 2010, 2015, 2020), but its accuracy is... disappointing. For later, maybe.

File: src/calliope_pathways/model_configs/italy_stationary/model.yaml [hunk 1]
```diff
@@ -12,7 +10,7 @@ config:
     # What version of Calliope this model is intended for
     calliope_version: 0.7.0
     # Activate pathways configuration
-    add_math: [../../math/pathways.yaml]
+    add_math: [../../math/pathways.yaml, math.yaml]
```
Comment (irm-codebase) [inline src/calliope_pathways/model_configs/italy_stationary/model.yaml:13]: I suspect we'll add more math, so let's move this into pathways/math/import_export.yaml ?

File: src/calliope_pathways/model_configs/italy_stationary/model.yaml [hunk 2]
```diff
@@ -70,27 +61,26 @@ data_sources:
...
-    rows: [techs]
-    columns: investsteps
-    add_dimensions:
-      parameters: available_initial_cap
-
   vintage_availability_techs:
     source: data_sources/investstep_series/available_vintages_techs.csv
     rows: techs
-    columns: [vintagesteps, investsteps]
+    columns: [investsteps, vintagesteps]
```
Comment (irm-codebase) [inline src/calliope_pathways/model_configs/italy_stationary/model.yaml:67]: image

Dammit, by bad.

File: src/calliope_pathways/model_configs/italy_stationary/model_config/techs.yaml [hunk 3]
```diff
@@ -248,18 +249,38 @@ techs: # Color scheme is Tol Qual Muted + some extras
     color: "#072486"
     base_tech: demand
     carrier_in: electricity
+    flow_cap_initial: 1e12  # very large number - it doesn't actually matter what this technology has as a capacity.
+    available_initial_cap: 1  # always available throughout the investment steps
```
Comment (irm-codebase) [inline src/calliope_pathways/model_configs/italy_stationary/model_config/techs.yaml:253]: Another mistake on my side. Thanks for catching it!

File: src/calliope_pathways/model_configs/italy_stationary/pre_processing/parse_lombardi.py [hunk 4]
```diff
@@ -96,27 +119,27 @@ def _location_yaml_to_df(yaml_data: dict, calliope_version: str = "0.6.8") -> pd
     return yaml_df
 
 
-def _build_tech_df(yml_path: str, calliope_version: str = "0.7") -> pd.DataFrame:
+def _get_lifetime_df(yml_path: str, calliope_version: str = "0.7") -> dict:
     """Converts a tech yaml into a searchable dataframe."""
```
Comment (irm-codebase) [inline src/calliope_pathways/model_configs/italy_stationary/pre_processing/parse_lombardi.py]: Update comment?

**Review by irm-codebase — DISMISSED** [review]
Apologies for the bugs Bryn. Things look great, just a couple of nice-to-haves.

Approved

File: src/calliope_pathways/model_configs/italy_stationary/model.yaml (hunk 1 above)
Comment (brynpickering) [reply src/calliope_pathways/model_configs/italy_stationary/model.yaml:13]: Since it is specific to the `italy` model, I wanted to stay with it.

Comment (irm-codebase) [reply src/calliope_pathways/model_configs/italy_stationary/model.yaml:13]: Ahhh, I get it. It's fine, then. We can rename this file if we introduce more math options.

**Review by irm-codebase — APPROVED** [review]

**Pull request merged**